nexis/lib/pages/home.dart

```diff
@@ -11,83 +19,256 @@ class Home extends StatefulWidget {
+        );
+      }
+    }
+  }
+
+  Future<void> listenToMessages() async {
+    FirestoreRead firestoreRead =
+        FirestoreRead(groupId: 'aNAgqEvRvtFLDmjw7Ivz');
+    try {
+      List<DocumentSnapshot> documents = await firestoreRead.initialFetch();
+      Set<String> processedIds = documents.map((doc) => doc.id).toSet();
+
+      messages.clear();
+
+      Timestamp? lastMessageTimestamp;
+      if (documents.isNotEmpty) {
+        for (var document in documents.reversed) {
+          messages.add(Message.fromDocument(document));
+        }
+
+        var lastMessageData = documents.last.data();
+        if (lastMessageData != null &&
+            lastMessageData is Map<String, dynamic>) {
+          lastMessageTimestamp = lastMessageData['timestamp'] as Timestamp?;
+        }
+      }
+
+      subscription = firestoreRead
+          .listenToNewMessages(lastMessageTimestamp)
+          .listen((querySnapshot) {
+        for (var docChange in querySnapshot.docChanges) {
+          if (docChange.type == DocumentChangeType.added) {
+            DocumentSnapshot newDocument = docChange.doc;
+            if (processedIds.contains(newDocument.id)) {
+              continue;
+            }
+            setState(() {
+              // Update the UI here
+              messages.add(Message.fromDocument(newDocument));
+            });
+          }
+        }
+      });
+
+      messagesLoaded = true;
+    } catch (error) {
+      throw Exception('Error listening to messages: $error');
+    }
+  }
 
   @override
   Widget build(BuildContext context) {
+    return FutureBuilder(
+      builder: (BuildContext context, AsyncSnapshot<void> snapshot) {
+        if (snapshot.hasError) {
+          return Text('Error: ${snapshot.error}');
+        } else {
+          return buildHomeScreen(context);
+        }
+      },
+    );
+  }
+
+  Widget buildHomeScreen(BuildContext context) {
     return Scaffold(
       backgroundColor: Theme.of(context).colorScheme.primary,
-      body: Center(
+      body: SafeArea(
         child: Column(
-          mainAxisAlignment: MainAxisAlignment.center,
+          crossAxisAlignment: CrossAxisAlignment.center,
+          children: [
+            buildMessagesListView(),
+            buildMessageInputArea(),
+          ],
+        ),
+      ),
+    );
+  }
+
+  Widget buildMessagesListView() {
+    return Expanded(
+      child: messages.isEmpty && messagesLoaded
+          ? const Center(
+              child: Text(
+                'No messages',
+                style: TextStyle(color: Colors.white),
+              ),
+            )
+          : ListView.builder(
+              padding: const EdgeInsets.symmetric(horizontal: 24, vertical: 24),
+              shrinkWrap: true,
+              itemCount: messages.length,
+              itemBuilder: (context, index) => buildMessageItem(index),
+            ),
+    );
+  }
+
+  Widget buildMessageItem(int index) {
```

Comment:
Probably should be a general widget since we wont use it later in just the home window.